File: airesources/Haskell/src/RandomBot.hs
module RandomBot
    ( algorithm
    , name
    ) where

import Types
import Classes

name :: String
name = "Random Haskell Bot"

algorithm :: RandomReader r => Int -> GameMap -> r [Move]
algorithm me g@(GameMap width height sites) =
   randomMoves
      [Location x y | x <- [1..width], y <- [1..height]] $
      fmap (isMy me) sites

randomMoves :: RandomReader r => [Location] -> [Bool] -> r [Move]
randomMoves l = traverse randMove . filter' l
   where
      filter' :: [a] -> [Bool] -> [a]
      filter' [] [] = []
      filter' (a:as) (True:bs)  = a : filter' as bs
      filter' (_:as) (False:bs) = filter' as bs
      randMove :: RandomReader r => Location -> r Move
      randMove location = do
         direction <- toEnum <$> randR 5
         return $ Move location direction

File: airesources/Elixir/lib/RandomBot.ex
defmodule RandomBot do   
    defp loop(id, game_map) do
        GameMap.locations(game_map)
        |> Enum.filter(&(GameMap.get_site(game_map, &1).owner == id))
        |> Enum.map(fn(location) -> {location, Enum.random(Direction.directions)} end)
        |> Networking.send_frame
        loop(id, Networking.get_frame(game_map))
    end

    def main do
        {id, game_map} = Networking.get_init
        Networking.send_init "ElixirBot"
        loop(id, Networking.get_frame(game_map))
    end
end